src/components/slider/slider.tsx

```diff
@@ -105,19 +107,31 @@ export class Slider {
     private mdcSlider: MDCSlider;
     private labelId: string;
     private helperTextId: string;
+    private observer: ResizeObserver;
 
     @State()
     private percentageClass: string;
 
     public constructor() {
+        const debounceTimeout = 200;
         this.inputHandler = this.inputHandler.bind(this);
         this.getContainerClassList = this.getContainerClassList.bind(this);
+        this.handleZoom = debounce(
+            this.handleZoom.bind(this),
+            debounceTimeout,
+            { leading: true, trailing: true }
+        );
+
         this.labelId = createRandomString();
         this.helperTextId = createRandomString();
     }
 
     public connectedCallback() {
         this.initialize();
+        this.observer = new ResizeObserver(this.handleResize) as any;
+        this.observer.observe(this.rootElement);
+        window.addEventListener('resize', this.handleZoom);
+        window.addEventListener('wheel', this.handleZoom);
```

Comment:
Hmm… here we're setting up listeners for `resize` and `wheel` events on `window`, but in `this.handleZoom`, we run `event.stopPropagation`. I guess it might not matter since we're catching the events in the bubbling phase, and presumably there's nothing left to bubble to after `window`, but it feels a bit dangerous to me to be running `stopPropagation` on those kinds of events… 🤔

Reply:
We can skip stopPropagation then. I think it makes the code a little simpler as well, since we can now use the same callback for the resize, wheel and resizeObserver callback.

In fact, I wrote.it on autopilot and it didn't occur to me that we were doing it on the window where it wouldn't propagate further up 😅